lib/membrane/rtp/demuxer/jitter_buffer.ex

```diff
@@ -0,0 +1,262 @@
+defmodule Membrane.RTP.Demuxer.JitterBuffer do
+  @moduledoc false
+
+  require Membrane.Logger
+
+  alias Membrane.{RTP, Time}
+  alias Membrane.RTP.Demuxer
+  alias Membrane.RTP.JitterBuffer.{BufferStore, Record}
+
+  @max_timestamp Bitwise.bsl(1, 32) - 1
+
+  @type pad_options ::
+          %{
+            stream_id: Demuxer.stream_id(),
+            clock_rate: RTP.clock_rate(),
+            use_jitter_buffer: boolean(),
+            jitter_buffer_latency: Membrane.Time.t()
+          }
+
+  defmodule State do
+    @moduledoc false
+
+    @type t :: %__MODULE__{
+            buffer_store: RTP.JitterBuffer.BufferStore.t(),
+            clock_rate: non_neg_integer() | nil,
+            latency: Membrane.Time.t() | nil,
+            initial_latency_waiting: boolean(),
+            initialization_time: Membrane.Time.t(),
+            max_latency_timer: reference() | nil,
+            timestamp_base: Membrane.Time.t(),
+            previous_timestamp: Membrane.Time.t()
+          }
+
+    @enforce_keys [
+      :initial_latency_waiting,
+      :initialization_time,
+      :timestamp_base,
+      :previous_timestamp
+    ]
+
+    defstruct @enforce_keys ++
+                [
+                  buffer_store: %RTP.JitterBuffer.BufferStore{},
+                  clock_rate: nil,
+                  latency: nil,
+                  max_latency_timer: nil
+                ]
+  end
+
+  @spec initialize_state_with_new_rtp_stream(
+          ExRTP.Packet.t(),
+          pad_options() | nil,
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: State.t()
+  def initialize_state_with_new_rtp_stream(packet, pad_options, payload_type_mapping) do
```

Comment:
how about renaming it to `new`, not passing `pad_options` at all, and requiring to always call `new` and then `complete_state_initialization_with_pad_options`?